### DISCUSSION TITLE
View - Technical Consensus

### DISCUSSION THREAD

**Discussion by jin-Pro** [githru/githru-vscode-ext#8] [View]
- feature 리스트 정리 ( https://github.com/orgs/githru/projects/2 )
  - feature 리스트 의논 discussion
  - 담당자 및 완료 시간
  - 우선순위
        
- 유지보수를 잘 하기 위해 공통으로 적용할 항목들에 대해 협의합니다.
  - eslint,prettier rule sets
  - test 전략 (테스트 범위, coverage, ...)
  - coding conventions
    - directory structure
    - file naming
  - bundler

- ...

수정해야할 부분과 추가해야할 부분에 대한 의견 반영 부탁드립니다!

**Comment by ytaek** [comment]
- Middleware(redux 같은?) vs context API 사용에 관한 토의,
- test 파일들도 naming convention 결정할 때 같이 논의하면 좋겠어요.
- 그 외 react 관련한 것들이 더 있으면? 시간되는대로 얘기해보면 좋겠습니다.
  ex) setState를 할 때 기본 타입을 null로 할지 undefined로 할지, empty value로 할지,... array는 undefined로 할지 [ ] 로 할지 등등?

**Comment by ytaek** [comment]
아참!! Material UI 등의 도움을 받을지? 이런 것도 얘기하면 좋겠어요!!
이쁘게 하려면 어떻게 해야 할까요? ㅜ.ㅜ

**Reply by ansrlm** [reply]
만약 사용하신다면 다음 예시들 비교해보시면 좋을 것 같습니다.
- Ant Design
- Semantic UI
- UIkit
- react-md
- MUI
- Lightning Design System
- Reakit
- Elastic UI
- Blueprint
